kolena/_experimental/dataset/_dataset.py

```diff
@@ -273,23 +279,70 @@ def _iter_dataset_raw(
     )
 
 
-def iter_dataset(
+def _iter_dataset(
     name: str,
+    commit: str = None,
     batch_size: int = BatchSize.LOAD_SAMPLES.value,
 ) -> Iterator[pd.DataFrame]:
     """
     Get an iterator over datapoints in the dataset.
     """
-    for df_batch in _iter_dataset_raw(name, batch_size):
+    for df_batch in _iter_dataset_raw(name, commit, batch_size):
         yield _to_deserialized_dataframe(df_batch, column=COL_DATAPOINT)
 
 
 def fetch_dataset(
     name: str,
+    commit: str = None,
     batch_size: int = BatchSize.LOAD_SAMPLES.value,
 ) -> pd.DataFrame:
     """
     Fetch an entire dataset given its name.
     """
-    df_batches = list(iter_dataset(name, batch_size))
+    df_batches = list(_iter_dataset(name, commit, batch_size))
     return pd.concat(df_batches, ignore_index=True) if df_batches else pd.DataFrame()
+
+
+def _list_commits(name: str, desc: bool = False, offset: int = 0, limit: int = 50) -> ListCommitHistoryResponse:
+    """
+    Invoke the list-commits api.
+    """
+    request = ListCommitHistoryRequest(name=name, desc=desc, offset=offset, limit=limit)
+    response = krequests.put(Path.LIST_COMMITS, json=asdict(request))
+    response.raise_for_status()
+    return from_dict(ListCommitHistoryResponse, response.json())
+
+
+def _iter_commits(
+    name: str,
+    desc: bool = False,
+    limit: int = None,
+    page_size: int = 50,
+) -> Iterator[Tuple[int, List[CommitData]]]:
+    """
+    Get an iterator over the commit history of the dataset.
+    """
+    initial_response = _list_commits(name, desc=desc, limit=page_size)
+    total_commit_count = initial_response.total_count
+    if not limit:
+        limit = total_commit_count
+    yield total_commit_count, initial_response.records
+    current_count = len(initial_response.records)
+    while current_count < min(limit, total_commit_count):
+        yield total_commit_count, _list_commits(name, desc=desc, offset=current_count, limit=page_size).records
+        current_count += len(initial_response.records)
+
+
+def fetch_commits(
```

Comment:
Nit: do we want a name that makes it more obvious this is for dataset? Maybe `fetch_dataset_history` or something? Debating if we should include `commit` in method name as it is a "new" concept that may need explanation.

Reply:
Good idea, will rename to `fetch_dataset_history`